Realm/RLMSyncManager.mm

```diff
@@ -164,20 +171,25 @@ + (void)_resetStateForTesting {
         }
     }
     [[s_sharedManager.activeUsers allValues] makeObjectsPerformSelector:@selector(logOut)];
+    NSURL *metadataURL = [s_sharedManager.fileManager fileURLForMetadata];
 
     // Reset the singleton.
+    s_testBundleID = nil;
     s_onceToken = 0;
```

Comment:
I wonder if it makes sense of the tests to swizzle `[RLMSyncManager sharedManager]` and just have each test instantiate an instance of the manager. I don't think the singleton state of `RLMSyncManager` is tied to the singleton state of any of the object store singletons.